src/main/java/org/elasticsearch/search/aggregations/AggregatorFactory.java

```diff
@@ -72,20 +77,20 @@ public AggregatorFactory parent() {
         return parent;
     }
 
+    protected abstract Aggregator createInternal(AggregationContext context, Aggregator parent, boolean collectsOnly0, Map<String, Object> metaData) throws IOException;
+
     /**
      * Creates the aggregator
      *
      * @param context               The aggregation context
      * @param parent                The parent aggregator (if this is a top level factory, the parent will be {@code null})
-     * @param expectedBucketsCount  If this is a sub-factory of another factory, this will indicate the number of bucket the parent aggregator
-     *                              may generate (this is an estimation only). For top level factories, this will always be 0
+     * @param collectsOnly0         If true then the created aggregator will only be collected with <tt>0</tt> as a bucket ordinal.
+     *                              Some factories can take advantage of this in order to return more optimized implementations.
      *
      * @return                      The created aggregator
      */
-    protected abstract Aggregator createInternal(AggregationContext context, Aggregator parent, long expectedBucketsCount, Map<String, Object> metaData);
-
-    public Aggregator create(AggregationContext context, Aggregator parent, long expectedBucketsCount) {
-        Aggregator aggregator = createInternal(context, parent, expectedBucketsCount, this.metaData);
+    public final Aggregator create(AggregationContext context, Aggregator parent, boolean collectsOnly0) throws IOException {
```

Comment:
Why do we have this given that all it does is return the result of createInternal? Is it just here for future-proofing the API?